mintpy/prep_aria.py

```diff
@@ -1,216 +1,290 @@
                         help='Name of the stack VRT file of unwrapped data.\n'+
-                              ' default: unwrapStack.vrt')
-    parser.add_argument('-c', '--coherence-stack-name', dest='cohStack', type=str,
+                             'default: unwrapStack.vrt')
+    parser.add_argument('-c','--coherence-stack-name', dest='cohStack', type=str,
                         default="cohStack.vrt",
                         help='Name of the stack VRT file of coherence data.\n'+
-                              'default: cohStack.vrt')
-    parser.add_argument('-l', '--conn-comp-name', dest='connCompStack', type=str,
+                             'default: cohStack.vrt')
+    parser.add_argument('-l','--conn-comp-name', dest='connCompStack', type=str,
                         default="connCompStack.vrt",
                         help='Name of the stack VRT file of connected component data.\n' +
-                              'default: connCompStack.vrt')
-    parser.add_argument('-i', '--incidence-angle', dest='incidenceAngle', type=str,
-                        required=True,
+                             'default: connCompStack.vrt')
+    parser.add_argument('-i','--incidence-angle', dest='incAngle', type=str, required=True,
                         help='Name of the incidence angle file')
-    parser.add_argument('-d', '--dem', dest='dem', type=str,
-                        required=True,
+    parser.add_argument('-d','--dem', dest='dem', type=str, required=True,
                         help='Name of the DEM file')
 
     return parser
 
+
 def cmd_line_parse(iargs = None):
     parser = create_parser()
     inps = parser.parse_args(args=iargs)
-
+    inps.stackDir = os.path.abspath(inps.stackDir)
     return inps
 
+
+####################################################################################
 def extract_metadata(stack):
 
+    print('extract metadata from {}'.format(stack))
     ds = gdal.Open(stack, gdal.GA_ReadOnly)
+    metaUnw = ds.GetRasterBand(1).GetMetadata("unwrappedPhase")
 
+    # copy over all metadata from unwrappedPhase
     meta = {}
+    for key, value in metaUnw.items():
+        if key not in ["Dates", "perpendicularBaseline"]:
+            meta[key] = value
+
+    meta["ANTENNA_SIDE"] = -1
     meta["PROCESSOR"] = "isce"
     meta["FILE_LENGTH"] = ds.RasterYSize
     meta["LENGTH"] = ds.RasterYSize
-    meta["ORBIT_DIRECTION"] = "DESCENDING"
+    meta["ORBIT_DIRECTION"] = "DESCENDING"    #check with ARIA-tools team
     meta["PLATFORM"] = "Sen"
-    meta["STARTING_RANGE"] = float(ds.GetRasterBand(1).GetMetadata("unwrappedPhase")['startRange'])
-    meta["WAVELENGTH"] = float(ds.GetRasterBand(1).GetMetadata("unwrappedPhase")['Wavelength (m)'])
+    meta["WAVELENGTH"] = float(meta["Wavelength (m)"])
     meta["WIDTH"] = ds.RasterXSize
     meta["NUMBER_OF_PAIRS"] = ds.RasterCount
 
-    # ARIA standard products currently don't number of range and
+    az_angle = float(meta["azimuthAngle"])
```

Comment:
@ehavazli and @dbekaert, this value `azimuthAngle = -8.77992802893209` does not seem like the azimuthAngle of the satellite orbit (band2 of los.rdr from ISCE/topsStack); but more like the azimuth angle of the LOS vector (from incLocal.rdr), could you confirm?

If we got the HEADING angle of the satellite orbit, the `ORBIT_DIRECTION` can be drived from here.

Reply:
azimuth angle in the VRT metadata is an average value. Azimuth angle information is extracted from the product. I am testing the new ariaTSsetup.py with the orbit direction information being included in the metadata of VRT.